<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>jimbox</class>
 <widget class="QMainWindow" name="jimbox">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>757</width>
    <height>510</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>jimbox</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout_2">
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="currentIndex">
       <number>1</number>
      </property>
      <widget class="QWidget" name="hatTab">
       <attribute name="title">
        <string>hat</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QGroupBox" name="thregroupBox">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>15</verstretch>
           </sizepolicy>
          </property>
          <property name="title">
           <string>Threshold</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_27">
           <item row="0" column="0" colspan="2">
            <widget class="QPushButton" name="otsuButton">
             <property name="text">
              <string>OTSU</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0" colspan="2">
            <widget class="QPushButton" name="entropyButton">
             <property name="text">
              <string>Entropy</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0" colspan="2">
            <widget class="QSlider" name="threSlider">
             <property name="maximum">
              <number>255</number>
             </property>
             <property name="value">
              <number>128</number>
             </property>
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <widget class="QSpinBox" name="threSpin">
             <property name="maximum">
              <number>255</number>
             </property>
            </widget>
           </item>
           <item row="4" column="0">
            <widget class="QCheckBox" name="inverseCheck">
             <property name="text">
              <string>inverse</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QGroupBox" name="savegroupBox">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>10</verstretch>
           </sizepolicy>
          </property>
          <property name="title">
           <string>Save</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_28">
           <item row="2" column="1">
            <widget class="QPushButton" name="saveButton">
             <property name="text">
              <string>SAVE</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0" colspan="2">
            <layout class="QHBoxLayout" name="horizontalLayout">
             <item>
              <widget class="QComboBox" name="saveChoice">
               <item>
                <property name="text">
                 <string>source image</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>histogram</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>gray image</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>binary image</string>
                </property>
               </item>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="saveFormat">
               <item>
                <property name="text">
                 <string>.jpg</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>.png</string>
                </property>
               </item>
              </widget>
             </item>
            </layout>
           </item>
           <item row="3" column="0" colspan="2">
            <widget class="QLabel" name="saveTip">
             <property name="text">
              <string/>
             </property>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QLabel" name="saveLabel1">
             <property name="text">
              <string>Choose image:</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="cafTab">
       <attribute name="title">
        <string>caf</string>
       </attribute>
       <widget class="QLabel" name="kernelLabel">
        <property name="geometry">
         <rect>
          <x>50</x>
          <y>250</y>
          <width>101</width>
          <height>81</height>
         </rect>
        </property>
        <property name="text">
         <string>TextLabel</string>
        </property>
        <property name="scaledContents">
         <bool>true</bool>
        </property>
       </widget>
       <widget class="QComboBox" name="filterChoice">
        <property name="geometry">
         <rect>
          <x>90</x>
          <y>350</y>
          <width>101</width>
          <height>22</height>
         </rect>
        </property>
        <item>
         <property name="text">
          <string>Correlation</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Convolution</string>
         </property>
        </item>
       </widget>
       <widget class="QLabel" name="opeLabel">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>350</y>
          <width>71</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>Operation:</string>
        </property>
       </widget>
       <widget class="QPushButton" name="filterButton">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="geometry">
         <rect>
          <x>50</x>
          <y>380</y>
          <width>75</width>
          <height>23</height>
         </rect>
        </property>
        <property name="text">
         <string>Filter</string>
        </property>
       </widget>
       <widget class="QGroupBox" name="groupBox">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>10</y>
          <width>181</width>
          <height>231</height>
         </rect>
        </property>
        <property name="title">
         <string>Filter choice</string>
        </property>
        <widget class="QRadioButton" name="manualfilButton">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>180</y>
           <width>89</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>Manual:</string>
         </property>
        </widget>
        <widget class="QRadioButton" name="medianButton">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>150</y>
           <width>89</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>Median</string>
         </property>
        </widget>
        <widget class="QLabel" name="label">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>20</y>
           <width>111</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>Edge Detection:</string>
         </property>
        </widget>
        <widget class="QRadioButton" name="gaussianButton">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>130</y>
           <width>81</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>Gaussian:</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_2">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>110</y>
           <width>101</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>Noise reduction:</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_3">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>40</y>
           <width>54</width>
           <height>12</height>
          </rect>
         </property>
         <property name="text">
          <string>Roberts:</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_4">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>60</y>
           <width>54</width>
           <height>12</height>
          </rect>
         </property>
         <property name="text">
          <string>Prewitt:</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_5">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>80</y>
           <width>54</width>
           <height>12</height>
          </rect>
         </property>
         <property name="text">
          <string>Sobel:</string>
         </property>
        </widget>
        <widget class="QRadioButton" name="robertsxButton">
         <property name="geometry">
          <rect>
           <x>70</x>
           <y>40</y>
           <width>31</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>Gx</string>
         </property>
        </widget>
        <widget class="QRadioButton" name="robertsyButton">
         <property name="geometry">
          <rect>
           <x>110</x>
           <y>40</y>
           <width>31</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>Gy</string>
         </property>
        </widget>
        <widget class="QRadioButton" name="prewittyButton">
         <property name="geometry">
          <rect>
           <x>110</x>
           <y>60</y>
           <width>31</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>Gy</string>
         </property>
        </widget>
        <widget class="QRadioButton" name="prewittxButton">
         <property name="geometry">
          <rect>
           <x>70</x>
           <y>60</y>
           <width>31</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>Gx</string>
         </property>
        </widget>
        <widget class="QRadioButton" name="sobelyButton">
         <property name="geometry">
          <rect>
           <x>110</x>
           <y>80</y>
           <width>31</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>Gy</string>
         </property>
        </widget>
        <widget class="QRadioButton" name="sobelxButton">
         <property name="geometry">
          <rect>
           <x>70</x>
           <y>80</y>
           <width>31</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>Gx</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_6">
         <property name="geometry">
          <rect>
           <x>90</x>
           <y>130</y>
           <width>21</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>σ:</string>
         </property>
        </widget>
        <widget class="QLineEdit" name="sigmaEdit">
         <property name="geometry">
          <rect>
           <x>110</x>
           <y>130</y>
           <width>31</width>
           <height>20</height>
          </rect>
         </property>
        </widget>
        <widget class="Line" name="line">
         <property name="geometry">
          <rect>
           <x>20</x>
           <y>100</y>
           <width>118</width>
           <height>3</height>
          </rect>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
        <widget class="Line" name="line_2">
         <property name="geometry">
          <rect>
           <x>20</x>
           <y>170</y>
           <width>118</width>
           <height>3</height>
          </rect>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
        <widget class="QLineEdit" name="kernelEdit">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>200</y>
           <width>141</width>
           <height>20</height>
          </rect>
         </property>
        </widget>
        <widget class="QPushButton" name="pushButton">
         <property name="geometry">
          <rect>
           <x>159</x>
           <y>200</y>
           <width>21</width>
           <height>23</height>
          </rect>
         </property>
         <property name="text">
          <string>ok</string>
         </property>
        </widget>
       </widget>
      </widget>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="displayBox">
      <property name="title">
       <string>Display</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_22">
       <item row="0" column="0">
        <layout class="QGridLayout" name="gridLayout_1">
         <item row="0" column="0" colspan="3">
          <widget class="QWidget" name="ltWidget" native="true">
           <property name="minimumSize">
            <size>
             <width>100</width>
             <height>100</height>
            </size>
           </property>
           <widget class="QLabel" name="ltLabel">
            <property name="geometry">
             <rect>
              <x>10</x>
              <y>10</y>
              <width>160</width>
              <height>160</height>
             </rect>
            </property>
            <property name="text">
             <string>siImage</string>
            </property>
            <property name="scaledContents">
             <bool>true</bool>
            </property>
           </widget>
          </widget>
         </item>
         <item row="1" column="0">
          <spacer name="horizontalSpacer_1">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>28</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="1" column="1">
          <widget class="QLabel" name="ltText">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>source image</string>
           </property>
           <property name="scaledContents">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <spacer name="horizontalSpacer_2">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>28</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
       <item row="0" column="1">
        <spacer name="horizontalSpacer_9">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Maximum</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>30</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="0" column="2">
        <layout class="QGridLayout" name="gridLayout_2">
         <item row="0" column="0" colspan="3">
          <widget class="QWidget" name="rtWidget" native="true">
           <property name="minimumSize">
            <size>
             <width>100</width>
             <height>100</height>
            </size>
           </property>
           <widget class="QLabel" name="rtLabel">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>0</y>
              <width>160</width>
              <height>160</height>
             </rect>
            </property>
            <property name="text">
             <string>hisImage</string>
            </property>
            <property name="scaledContents">
             <bool>true</bool>
            </property>
           </widget>
          </widget>
         </item>
         <item row="1" column="0">
          <spacer name="horizontalSpacer_3">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>28</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="1" column="1">
          <widget class="QLabel" name="rtText">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>histogram</string>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <spacer name="horizontalSpacer_4">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>28</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
       <item row="1" column="0">
        <layout class="QGridLayout" name="gridLayout_3">
         <item row="0" column="0" colspan="3">
          <widget class="QWidget" name="lbWidget" native="true">
           <property name="minimumSize">
            <size>
             <width>100</width>
             <height>100</height>
            </size>
           </property>
           <widget class="QLabel" name="lbLabel">
            <property name="geometry">
             <rect>
              <x>10</x>
              <y>10</y>
              <width>160</width>
              <height>160</height>
             </rect>
            </property>
            <property name="text">
             <string>giImage</string>
            </property>
            <property name="scaledContents">
             <bool>true</bool>
            </property>
           </widget>
          </widget>
         </item>
         <item row="1" column="0">
          <spacer name="horizontalSpacer_5">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>28</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="1" column="1">
          <widget class="QLabel" name="lbText">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>gray image</string>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <spacer name="horizontalSpacer_6">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>28</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
       <item row="1" column="1">
        <spacer name="horizontalSpacer_10">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Maximum</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>30</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="1" column="2">
        <layout class="QGridLayout" name="gridLayout_4">
         <item row="0" column="0" colspan="3">
          <widget class="QWidget" name="rbWidget" native="true">
           <property name="minimumSize">
            <size>
             <width>100</width>
             <height>100</height>
            </size>
           </property>
           <widget class="QLabel" name="rbLabel">
            <property name="geometry">
             <rect>
              <x>10</x>
              <y>10</y>
              <width>160</width>
              <height>160</height>
             </rect>
            </property>
            <property name="text">
             <string>biImage</string>
            </property>
            <property name="scaledContents">
             <bool>true</bool>
            </property>
           </widget>
          </widget>
         </item>
         <item row="1" column="0">
          <spacer name="horizontalSpacer_7">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>28</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="1" column="1">
          <widget class="QLabel" name="rbText">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>binary image</string>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <spacer name="horizontalSpacer_8">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>28</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>757</width>
     <height>23</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_File">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="action_Open"/>
   </widget>
   <addaction name="menu_File"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="action_Open">
   <property name="text">
    <string>&amp;Open</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>threSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>threSpin</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>130</x>
     <y>181</y>
    </hint>
    <hint type="destinationlabel">
     <x>130</x>
     <y>196</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>threSpin</sender>
   <signal>valueChanged(int)</signal>
   <receiver>threSlider</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>106</x>
     <y>205</y>
    </hint>
    <hint type="destinationlabel">
     <x>80</x>
     <y>176</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
